
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="https://cmungall.github.io/linkml-phenopackets/" rel="canonical"/>
<link href="assets/images/favicon.png" rel="icon"/>
<meta content="mkdocs-1.4.2, mkdocs-material-8.5.10" name="generator"/>
<title>phenopackets in linkml EXPERIMENTAL DEMO</title>
<link href="assets/stylesheets/main.472b142f.min.css" rel="stylesheet"/>
<link href="assets/stylesheets/palette.08040f6c.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
</head>
<body data-md-color-accent="amber" data-md-color-primary="teal" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#phenopackets">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="phenopackets in linkml EXPERIMENTAL DEMO" class="md-header__button md-logo" data-md-component="logo" href="." title="phenopackets in linkml EXPERIMENTAL DEMO">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            phenopackets in linkml EXPERIMENTAL DEMO
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Index
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="amber" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="teal" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 2a7 7 0 0 0-7 7c0 2.38 1.19 4.47 3 5.74V17a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1v-2.26c1.81-1.27 3-3.36 3-5.74a7 7 0 0 0-7-7M9 21a1 1 0 0 0 1 1h4a1 1 0 0 0 1-1v-1H9v1Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="amber" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="teal" data-md-color-scheme="slate" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 2a7 7 0 0 1 7 7c0 2.38-1.19 4.47-3 5.74V17a1 1 0 0 1-1 1H9a1 1 0 0 1-1-1v-2.26C6.19 13.47 5 11.38 5 9a7 7 0 0 1 7-7M9 21v-1h6v1a1 1 0 0 1-1 1h-4a1 1 0 0 1-1-1m3-17a5 5 0 0 0-5 5c0 2.05 1.23 3.81 3 4.58V16h4v-2.42c1.77-.77 3-2.53 3-4.58a5 5 0 0 0-5-5Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/cmungall/linkml-phenopackets/" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="phenopackets in linkml EXPERIMENTAL DEMO" class="md-nav__button md-logo" data-md-component="logo" href="." title="phenopackets in linkml EXPERIMENTAL DEMO">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"></path></svg>
</a>
    phenopackets in linkml EXPERIMENTAL DEMO
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/cmungall/linkml-phenopackets/" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="home/">
        Home
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" data-md-toggle="toc" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Index
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href=".">
        Index
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#schema-diagram">
    Schema Diagram
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#classes">
    Classes
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#slots">
    Slots
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#enumerations">
    Enumerations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#types">
    Types
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#subsets">
    Subsets
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#schema-diagram">
    Schema Diagram
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#classes">
    Classes
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#slots">
    Slots
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#enumerations">
    Enumerations
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#types">
    Types
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#subsets">
    Subsets
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<a class="md-content__button md-icon" href="https://github.com/cmungall/linkml-phenopackets/edit/master/docs/index.md" title="Edit this page">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg>
</a>
<h1 id="phenopackets">phenopackets</h1>
<p>Automatic translation of phenopackets protobuf to LinkML. Status: EXPERIMENTAL.</p>
<p>URI: https://w3id.org/linkml/phenopackets/phenopackets
Name: phenopackets</p>
<h2 id="schema-diagram">Schema Diagram</h2>
<div class="mermaid">erDiagram
Phenopacket {
    string id  
}
Individual {
    stringList alternateIds  
    string dateOfBirth  
    string id  
    KaryotypicSex karyotypicSex  
    Sex sex  
}
VitalStatus {
    Status status  
    integer survivalTimeInDays  
}
TimeElement {
    string timestamp  
}
OntologyClass {
    string id  
    string label  
}
TimeInterval {
    string end  
    string start  
}
GestationalAge {
    integer days  
    integer weeks  
}
AgeRange {

}
Age {
    string iso8601duration  
}
PhenotypicFeature {
    string description  
    boolean excluded  
}
Evidence {

}
ExternalReference {
    string description  
    string id  
    string reference  
}
MetaData {
    string created  
    string createdBy  
    string phenopacketSchemaVersion  
    string submittedBy  
}
Update {
    string comment  
    string timestamp  
    string updatedBy  
}
Resource {
    string id  
    string iriPrefix  
    string name  
    string namespacePrefix  
    string url  
    string version  
}
MedicalAction {

}
Treatment {
    DrugType drugType  
}
DoseInterval {

}
Quantity {
    double value  
}
ReferenceRange {
    double high  
    double low  
}
TherapeuticRegimen {
    RegimenStatus regimenStatus  
}
RadiationTherapy {
    integer dosage  
    integer fractions  
}
Procedure {

}
Measurement {
    string description  
}
Value {

}
ComplexValue {

}
TypedQuantity {

}
Interpretation {
    string id  
    ProgressStatus progressStatus  
    string summary  
}
Diagnosis {

}
GenomicInterpretation {
    InterpretationStatus interpretationStatus  
    string subjectOrBiosampleId  
}
VariantInterpretation {
    AcmgPathogenicityClassification acmgPathogenicityClassification  
    TherapeuticActionability therapeuticActionability  
}
VariationDescriptor {
    stringList alternateLabels  
    string description  
    string id  
    string label  
    MoleculeContext moleculeContext  
    string vrsRefAlleleSeq  
    stringList xrefs  
}
VcfRecord {
    string alt  
    string chrom  
    string filter  
    string genomeAssembly  
    string id  
    string info  
    integer pos  
    string qual  
    string ref  
}
Variation {

}
VariationSet {

}
Text {
    string definition  
    string id  
}
Haplotype {

}
CopyNumber {
    string curie  
    string id  
}
RepeatedSequenceExpression {

}
Number {
    integer value  
}
LiteralSequenceExpression {
    string sequence  
}
IndefiniteRange {
    string comparator  
    integer value  
}
DerivedSequenceExpression {
    boolean reverseComplement  
}
SequenceLocation {
    string id  
    string sequenceId  
}
SequenceInterval {

}
DefiniteRange {
    integer max  
    integer min  
}
Gene {
    string geneId  
}
Allele {
    string curie  
    string id  
}
ChromosomeLocation {
    string chr  
    string id  
    string speciesId  
}
CytobandInterval {
    string end  
    string start  
}
GeneDescriptor {
    stringList alternateIds  
    stringList alternateSymbols  
    string description  
    string symbol  
    string valueId  
    stringList xrefs  
}
Extension {
    string name  
}
Any {
    string typeUrl  
    string value  
}
Expression {
    string syntax  
    string value  
    string version  
}
File {
    string uri  
}
Dictionary {

}
Disease {
    boolean excluded  
}
Biosample {
    string derivedFromId  
    string description  
    string id  
    string individualId  
}

Phenopacket ||--}o Biosample : "biosamples"
Phenopacket ||--}o Disease : "diseases"
Phenopacket ||--}o File : "files"
Phenopacket ||--}o Interpretation : "interpretations"
Phenopacket ||--}o Measurement : "measurements"
Phenopacket ||--}o MedicalAction : "medicalActions"
Phenopacket ||--|| MetaData : "metaData"
Phenopacket ||--}o PhenotypicFeature : "phenotypicFeatures"
Phenopacket ||--|o Individual : "subject"
Individual ||--|o OntologyClass : "gender"
Individual ||--|o OntologyClass : "taxonomy"
Individual ||--|o TimeElement : "timeAtLastEncounter"
Individual ||--|o VitalStatus : "vitalStatus"
VitalStatus ||--|o OntologyClass : "causeOfDeath"
VitalStatus ||--|o TimeElement : "timeOfDeath"
TimeElement ||--|o Age : "age"
TimeElement ||--|o AgeRange : "ageRange"
TimeElement ||--|o GestationalAge : "gestationalAge"
TimeElement ||--|o TimeInterval : "interval"
TimeElement ||--|o OntologyClass : "ontologyClass"
AgeRange ||--|o Age : "end"
AgeRange ||--|o Age : "start"
PhenotypicFeature ||--}o Evidence : "evidence"
PhenotypicFeature ||--}o OntologyClass : "modifiers"
PhenotypicFeature ||--|o TimeElement : "onset"
PhenotypicFeature ||--|o TimeElement : "resolution"
PhenotypicFeature ||--|o OntologyClass : "severity"
PhenotypicFeature ||--|o OntologyClass : "type"
Evidence ||--|o OntologyClass : "evidenceCode"
Evidence ||--|o ExternalReference : "reference"
MetaData ||--}o ExternalReference : "externalReferences"
MetaData ||--}o Resource : "resources"
MetaData ||--}o Update : "updates"
MedicalAction ||--}o OntologyClass : "adverseEvents"
MedicalAction ||--|o Procedure : "procedure"
MedicalAction ||--|o RadiationTherapy : "radiationTherapy"
MedicalAction ||--|o OntologyClass : "responseToTreatment"
MedicalAction ||--|o TherapeuticRegimen : "therapeuticRegimen"
MedicalAction ||--|o Treatment : "treatment"
MedicalAction ||--|o OntologyClass : "treatmentIntent"
MedicalAction ||--|o OntologyClass : "treatmentTarget"
MedicalAction ||--|o OntologyClass : "treatmentTerminationReason"
Treatment ||--|o OntologyClass : "agent"
Treatment ||--|o Quantity : "cumulativeDose"
Treatment ||--}o DoseInterval : "doseIntervals"
Treatment ||--|o OntologyClass : "routeOfAdministration"
DoseInterval ||--|o TimeInterval : "interval"
DoseInterval ||--|o Quantity : "quantity"
DoseInterval ||--|o OntologyClass : "scheduleFrequency"
Quantity ||--|o ReferenceRange : "referenceRange"
Quantity ||--|o OntologyClass : "unit"
ReferenceRange ||--|o OntologyClass : "unit"
TherapeuticRegimen ||--|o TimeElement : "endTime"
TherapeuticRegimen ||--|o ExternalReference : "externalReference"
TherapeuticRegimen ||--|o OntologyClass : "ontologyClass"
TherapeuticRegimen ||--|o TimeElement : "startTime"
RadiationTherapy ||--|| OntologyClass : "bodySite"
RadiationTherapy ||--|| OntologyClass : "modality"
Procedure ||--|o OntologyClass : "bodySite"
Procedure ||--|o OntologyClass : "code"
Procedure ||--|o TimeElement : "performed"
Measurement ||--|o OntologyClass : "assay"
Measurement ||--|o ComplexValue : "complexValue"
Measurement ||--|o Procedure : "procedure"
Measurement ||--|o TimeElement : "timeObserved"
Measurement ||--|o Value : "value"
Value ||--|o OntologyClass : "ontologyClass"
Value ||--|o Quantity : "quantity"
ComplexValue ||--}o TypedQuantity : "typedQuantities"
TypedQuantity ||--|o Quantity : "quantity"
TypedQuantity ||--|o OntologyClass : "type"
Interpretation ||--|o Diagnosis : "diagnosis"
Diagnosis ||--|o OntologyClass : "disease"
Diagnosis ||--}o GenomicInterpretation : "genomicInterpretations"
GenomicInterpretation ||--|o GeneDescriptor : "gene"
GenomicInterpretation ||--|o VariantInterpretation : "variantInterpretation"
VariantInterpretation ||--|o VariationDescriptor : "variationDescriptor"
VariationDescriptor ||--|o OntologyClass : "allelicState"
VariationDescriptor ||--}o Expression : "expressions"
VariationDescriptor ||--}o Extension : "extensions"
VariationDescriptor ||--|o GeneDescriptor : "geneContext"
VariationDescriptor ||--|o OntologyClass : "structuralType"
VariationDescriptor ||--|o Variation : "variation"
VariationDescriptor ||--|o VcfRecord : "vcfRecord"
Variation ||--|o Allele : "allele"
Variation ||--|o CopyNumber : "copyNumber"
Variation ||--|o Haplotype : "haplotype"
Variation ||--|o Text : "text"
Variation ||--|o VariationSet : "variationSet"
CopyNumber ||--|o Allele : "allele"
CopyNumber ||--|o DefiniteRange : "definiteRange"
CopyNumber ||--|o DerivedSequenceExpression : "derivedSequenceExpression"
CopyNumber ||--|o Gene : "gene"
CopyNumber ||--|o Haplotype : "haplotype"
CopyNumber ||--|o IndefiniteRange : "indefiniteRange"
CopyNumber ||--|o LiteralSequenceExpression : "literalSequenceExpression"
CopyNumber ||--|o Number : "number"
CopyNumber ||--|o RepeatedSequenceExpression : "repeatedSequenceExpression"
RepeatedSequenceExpression ||--|o DefiniteRange : "definiteRange"
RepeatedSequenceExpression ||--|o DerivedSequenceExpression : "derivedSequenceExpression"
RepeatedSequenceExpression ||--|o IndefiniteRange : "indefiniteRange"
RepeatedSequenceExpression ||--|o LiteralSequenceExpression : "literalSequenceExpression"
RepeatedSequenceExpression ||--|o Number : "number"
DerivedSequenceExpression ||--|o SequenceLocation : "location"
SequenceLocation ||--|o SequenceInterval : "sequenceInterval"
SequenceInterval ||--|o DefiniteRange : "endDefiniteRange"
SequenceInterval ||--|o IndefiniteRange : "endIndefiniteRange"
SequenceInterval ||--|o Number : "endNumber"
SequenceInterval ||--|o DefiniteRange : "startDefiniteRange"
SequenceInterval ||--|o IndefiniteRange : "startIndefiniteRange"
SequenceInterval ||--|o Number : "startNumber"
Allele ||--|o ChromosomeLocation : "chromosomeLocation"
Allele ||--|o DerivedSequenceExpression : "derivedSequenceExpression"
Allele ||--|o LiteralSequenceExpression : "literalSequenceExpression"
Allele ||--|o RepeatedSequenceExpression : "repeatedSequenceExpression"
Allele ||--|o SequenceLocation : "sequenceLocation"
ChromosomeLocation ||--|o CytobandInterval : "interval"
Extension ||--}o Any : "value"
File ||--|o Dictionary : "fileAttributes"
File ||--|o Dictionary : "individualToFileIdentifiers"
Disease ||--}o OntologyClass : "clinicalTnmFinding"
Disease ||--}o OntologyClass : "diseaseStage"
Disease ||--|o OntologyClass : "laterality"
Disease ||--|o TimeElement : "onset"
Disease ||--|o OntologyClass : "primarySite"
Disease ||--|o TimeElement : "resolution"
Disease ||--|o OntologyClass : "term"
Biosample ||--}o OntologyClass : "diagnosticMarkers"
Biosample ||--}o File : "files"
Biosample ||--|o OntologyClass : "histologicalDiagnosis"
Biosample ||--|o OntologyClass : "materialSample"
Biosample ||--}o Measurement : "measurements"
Biosample ||--|o OntologyClass : "pathologicalStage"
Biosample ||--}o OntologyClass : "pathologicalTnmFinding"
Biosample ||--}o PhenotypicFeature : "phenotypicFeatures"
Biosample ||--|o Procedure : "procedure"
Biosample ||--|o OntologyClass : "sampleProcessing"
Biosample ||--|o OntologyClass : "sampleStorage"
Biosample ||--|o OntologyClass : "sampleType"
Biosample ||--|o OntologyClass : "sampledTissue"
Biosample ||--|o OntologyClass : "taxonomy"
Biosample ||--|o TimeElement : "timeOfCollection"
Biosample ||--|o OntologyClass : "tumorGrade"
Biosample ||--|o OntologyClass : "tumorProgression"

</div>
<h2 id="classes">Classes</h2>
<table>
<thead>
<tr>
<th>Class</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="Abundance/">Abundance</a></td>
<td></td>
</tr>
<tr>
<td><a href="Age/">Age</a></td>
<td>See http://build</td>
</tr>
<tr>
<td><a href="AgeRange/">AgeRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="Allele/">Allele</a></td>
<td></td>
</tr>
<tr>
<td><a href="Any/">Any</a></td>
<td><code>Any</code> contains an arbitrary serialized protocol buffer message along with a U...</td>
</tr>
<tr>
<td><a href="Biosample/">Biosample</a></td>
<td>A Biosample refers to a unit of biological material from which the substrate ...</td>
</tr>
<tr>
<td><a href="ChromosomeLocation/">ChromosomeLocation</a></td>
<td></td>
</tr>
<tr>
<td><a href="Cohort/">Cohort</a></td>
<td>A group of individuals related in some phenotypic or genotypic aspect</td>
</tr>
<tr>
<td><a href="ComplexValue/">ComplexValue</a></td>
<td></td>
</tr>
<tr>
<td><a href="CopyNumber/">CopyNumber</a></td>
<td></td>
</tr>
<tr>
<td><a href="CytobandInterval/">CytobandInterval</a></td>
<td></td>
</tr>
<tr>
<td><a href="DefiniteRange/">DefiniteRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="DerivedSequenceExpression/">DerivedSequenceExpression</a></td>
<td></td>
</tr>
<tr>
<td><a href="Diagnosis/">Diagnosis</a></td>
<td></td>
</tr>
<tr>
<td><a href="Dictionary/">Dictionary</a></td>
<td></td>
</tr>
<tr>
<td><a href="Disease/">Disease</a></td>
<td>Message to indicate a disease (diagnosis) and its recorded onset</td>
</tr>
<tr>
<td><a href="DoseInterval/">DoseInterval</a></td>
<td>e</td>
</tr>
<tr>
<td><a href="Evidence/">Evidence</a></td>
<td>FHIR mapping: Condition</td>
</tr>
<tr>
<td><a href="Expression/">Expression</a></td>
<td>https://vrsatile</td>
</tr>
<tr>
<td><a href="Extension/">Extension</a></td>
<td>https://vrsatile</td>
</tr>
<tr>
<td><a href="ExternalReference/">ExternalReference</a></td>
<td>FHIR mapping: Reference (https://www</td>
</tr>
<tr>
<td><a href="Family/">Family</a></td>
<td>Phenotype, sample and pedigree data required for a genomic diagnosis</td>
</tr>
<tr>
<td><a href="Feature/">Feature</a></td>
<td></td>
</tr>
<tr>
<td><a href="File/">File</a></td>
<td></td>
</tr>
<tr>
<td><a href="Gene/">Gene</a></td>
<td></td>
</tr>
<tr>
<td><a href="GeneDescriptor/">GeneDescriptor</a></td>
<td>https://vrsatile</td>
</tr>
<tr>
<td><a href="GenomicInterpretation/">GenomicInterpretation</a></td>
<td>A statement about the contribution of a genomic element towards the observed ...</td>
</tr>
<tr>
<td><a href="GestationalAge/">GestationalAge</a></td>
<td></td>
</tr>
<tr>
<td><a href="Haplotype/">Haplotype</a></td>
<td></td>
</tr>
<tr>
<td><a href="IndefiniteRange/">IndefiniteRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="Individual/">Individual</a></td>
<td>An individual (or subject) typically corresponds to an individual human or an...</td>
</tr>
<tr>
<td><a href="Interpretation/">Interpretation</a></td>
<td></td>
</tr>
<tr>
<td><a href="LiteralSequenceExpression/">LiteralSequenceExpression</a></td>
<td></td>
</tr>
<tr>
<td><a href="Location/">Location</a></td>
<td></td>
</tr>
<tr>
<td><a href="Measurement/">Measurement</a></td>
<td>FHIR mapping: Observation (https://www</td>
</tr>
<tr>
<td><a href="MedicalAction/">MedicalAction</a></td>
<td>medication, procedure, other actions taken for clinical management</td>
</tr>
<tr>
<td><a href="Member/">Member</a></td>
<td></td>
</tr>
<tr>
<td><a href="MetaData/">MetaData</a></td>
<td></td>
</tr>
<tr>
<td><a href="MolecularVariation/">MolecularVariation</a></td>
<td></td>
</tr>
<tr>
<td><a href="Number/">Number</a></td>
<td></td>
</tr>
<tr>
<td><a href="OntologyClass/">OntologyClass</a></td>
<td>A class (aka term, concept) in an ontology</td>
</tr>
<tr>
<td><a href="Pedigree/">Pedigree</a></td>
<td>https://software</td>
</tr>
<tr>
<td><a href="Person/">Person</a></td>
<td></td>
</tr>
<tr>
<td><a href="Phenopacket/">Phenopacket</a></td>
<td>An anonymous phenotypic description of an individual or biosample with potent...</td>
</tr>
<tr>
<td><a href="PhenotypicFeature/">PhenotypicFeature</a></td>
<td>An individual phenotypic feature, observed as either present or absent (negat...</td>
</tr>
<tr>
<td><a href="Procedure/">Procedure</a></td>
<td>A clinical procedure performed on a subject</td>
</tr>
<tr>
<td><a href="Quantity/">Quantity</a></td>
<td></td>
</tr>
<tr>
<td><a href="RadiationTherapy/">RadiationTherapy</a></td>
<td>RadiationTherapy</td>
</tr>
<tr>
<td><a href="ReferenceRange/">ReferenceRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="RepeatedSequenceExpression/">RepeatedSequenceExpression</a></td>
<td></td>
</tr>
<tr>
<td><a href="Resource/">Resource</a></td>
<td>Description of an external resource used for referencing an object</td>
</tr>
<tr>
<td><a href="SequenceExpression/">SequenceExpression</a></td>
<td></td>
</tr>
<tr>
<td><a href="SequenceInterval/">SequenceInterval</a></td>
<td></td>
</tr>
<tr>
<td><a href="SequenceLocation/">SequenceLocation</a></td>
<td></td>
</tr>
<tr>
<td><a href="SequenceState/">SequenceState</a></td>
<td></td>
</tr>
<tr>
<td><a href="SimpleInterval/">SimpleInterval</a></td>
<td></td>
</tr>
<tr>
<td><a href="SystemicVariation/">SystemicVariation</a></td>
<td></td>
</tr>
<tr>
<td><a href="Text/">Text</a></td>
<td></td>
</tr>
<tr>
<td><a href="TherapeuticRegimen/">TherapeuticRegimen</a></td>
<td>ARGO mapping radiation::radiation_therapy_type (missing)</td>
</tr>
<tr>
<td><a href="TimeElement/">TimeElement</a></td>
<td></td>
</tr>
<tr>
<td><a href="TimeInterval/">TimeInterval</a></td>
<td></td>
</tr>
<tr>
<td><a href="Timestamp/">Timestamp</a></td>
<td>A Timestamp represents a point in time independent of any time zone or local ...</td>
</tr>
<tr>
<td><a href="Treatment/">Treatment</a></td>
<td>ARGO mapping treatment::is_primary_treatment (missing) treatment with an agen...</td>
</tr>
<tr>
<td><a href="TypedQuantity/">TypedQuantity</a></td>
<td>For complex measurements, such as blood pressure where more than one componen...</td>
</tr>
<tr>
<td><a href="Update/">Update</a></td>
<td>Information about when an update to a record occurred, who or what made the u...</td>
</tr>
<tr>
<td><a href="UtilityVariation/">UtilityVariation</a></td>
<td></td>
</tr>
<tr>
<td><a href="Value/">Value</a></td>
<td></td>
</tr>
<tr>
<td><a href="VariantInterpretation/">VariantInterpretation</a></td>
<td></td>
</tr>
<tr>
<td><a href="Variation/">Variation</a></td>
<td></td>
</tr>
<tr>
<td><a href="VariationDescriptor/">VariationDescriptor</a></td>
<td></td>
</tr>
<tr>
<td><a href="VariationSet/">VariationSet</a></td>
<td></td>
</tr>
<tr>
<td><a href="VcfRecord/">VcfRecord</a></td>
<td></td>
</tr>
<tr>
<td><a href="VitalStatus/">VitalStatus</a></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Slot</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="acmgPathogenicityClassification/">acmgPathogenicityClassification</a></td>
<td></td>
</tr>
<tr>
<td><a href="adverseEvents/">adverseEvents</a></td>
<td>ARGO mapping treatment::adverse_events</td>
</tr>
<tr>
<td><a href="affectedStatus/">affectedStatus</a></td>
<td></td>
</tr>
<tr>
<td><a href="age/">age</a></td>
<td></td>
</tr>
<tr>
<td><a href="agent/">agent</a></td>
<td></td>
</tr>
<tr>
<td><a href="ageRange/">ageRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="allele/">allele</a></td>
<td></td>
</tr>
<tr>
<td><a href="allelicState/">allelicState</a></td>
<td>We RECOMMEND that the allelic_state of variant be described by terms from the...</td>
</tr>
<tr>
<td><a href="alt/">alt</a></td>
<td></td>
</tr>
<tr>
<td><a href="alternateIds/">alternateIds</a></td>
<td>An optional list of alternative identifiers for this individual</td>
</tr>
<tr>
<td><a href="alternateLabels/">alternateLabels</a></td>
<td>Common aliases for a variant, e</td>
</tr>
<tr>
<td><a href="alternateSymbols/">alternateSymbols</a></td>
<td>Takes the place of alternate_labels field in a generic descriptor</td>
</tr>
<tr>
<td><a href="assay/">assay</a></td>
<td>An ontology class which describes the assay used to produce the measurement</td>
</tr>
<tr>
<td><a href="biosamples/">biosamples</a></td>
<td>Biosample(s) derived from the patient or a collection of biosamples in isolat...</td>
</tr>
<tr>
<td><a href="bodySite/">bodySite</a></td>
<td>FHIR mapping: Procedure</td>
</tr>
<tr>
<td><a href="causeOfDeath/">causeOfDeath</a></td>
<td>ARGO mapping donor::cause_of_death</td>
</tr>
<tr>
<td><a href="chr/">chr</a></td>
<td></td>
</tr>
<tr>
<td><a href="chrom/">chrom</a></td>
<td></td>
</tr>
<tr>
<td><a href="chromosomeLocation/">chromosomeLocation</a></td>
<td></td>
</tr>
<tr>
<td><a href="clinicalTnmFinding/">clinicalTnmFinding</a></td>
<td>Cancer findings in the TNM system that is relevant to the diagnosis of cancer</td>
</tr>
<tr>
<td><a href="code/">code</a></td>
<td>FHIR mapping: Procedure</td>
</tr>
<tr>
<td><a href="comment/">comment</a></td>
<td>Textual comment about the changes made to the content and/or reason for the u...</td>
</tr>
<tr>
<td><a href="comparator/">comparator</a></td>
<td></td>
</tr>
<tr>
<td><a href="complexValue/">complexValue</a></td>
<td></td>
</tr>
<tr>
<td><a href="consanguinousParents/">consanguinousParents</a></td>
<td>flag to indicate that the parents of the proband are consanguinous</td>
</tr>
<tr>
<td><a href="copyNumber/">copyNumber</a></td>
<td></td>
</tr>
<tr>
<td><a href="created/">created</a></td>
<td>ISO8601 UTC timestamp for when this phenopacket was created in ISO "2018-03-0...</td>
</tr>
<tr>
<td><a href="createdBy/">createdBy</a></td>
<td>some kind of identifier for the contributor/ program ARGO sample_registration...</td>
</tr>
<tr>
<td><a href="cumulativeDose/">cumulativeDose</a></td>
<td>ARGO mapping chemotherapy::cumulative_drug_dosage</td>
</tr>
<tr>
<td><a href="curie/">curie</a></td>
<td></td>
</tr>
<tr>
<td><a href="dateOfBirth/">dateOfBirth</a></td>
<td>The date of birth of the individual as an ISO8601 UTC timestamp - rounded dow...</td>
</tr>
<tr>
<td><a href="days/">days</a></td>
<td></td>
</tr>
<tr>
<td><a href="definiteRange/">definiteRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="definition/">definition</a></td>
<td></td>
</tr>
<tr>
<td><a href="derivedFromId/">derivedFromId</a></td>
<td>The id of the parent biosample this biosample was derived from</td>
</tr>
<tr>
<td><a href="derivedSequenceExpression/">derivedSequenceExpression</a></td>
<td></td>
</tr>
<tr>
<td><a href="description/">description</a></td>
<td></td>
</tr>
<tr>
<td><a href="diagnosis/">diagnosis</a></td>
<td>The diagnosis made in this interpretation</td>
</tr>
<tr>
<td><a href="diagnosticMarkers/">diagnosticMarkers</a></td>
<td>Clinically relevant bio markers</td>
</tr>
<tr>
<td><a href="disease/">disease</a></td>
<td>The disease/condition assigned to the diagnosis</td>
</tr>
<tr>
<td><a href="diseases/">diseases</a></td>
<td>Field for disease identifiers - could be used for listing either diagnosed or...</td>
</tr>
<tr>
<td><a href="diseaseStage/">diseaseStage</a></td>
<td>Disease staging, the extent to which a disease has developed</td>
</tr>
<tr>
<td><a href="dosage/">dosage</a></td>
<td>The total dose given in units of Gray (Gy)</td>
</tr>
<tr>
<td><a href="doseIntervals/">doseIntervals</a></td>
<td></td>
</tr>
<tr>
<td><a href="drugType/">drugType</a></td>
<td></td>
</tr>
<tr>
<td><a href="end/">end</a></td>
<td></td>
</tr>
<tr>
<td><a href="endDefiniteRange/">endDefiniteRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="endIndefiniteRange/">endIndefiniteRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="endNumber/">endNumber</a></td>
<td></td>
</tr>
<tr>
<td><a href="endTime/">endTime</a></td>
<td>end time can be empty which would indicate ongoing</td>
</tr>
<tr>
<td><a href="evidence/">evidence</a></td>
<td>Evidences for how the phenotype was determined</td>
</tr>
<tr>
<td><a href="evidenceCode/">evidenceCode</a></td>
<td>The encoded evidence type using, for example the Evidence &amp; Conclusion Ontolo...</td>
</tr>
<tr>
<td><a href="excluded/">excluded</a></td>
<td>Flag to indicate whether the phenotype was observed or not</td>
</tr>
<tr>
<td><a href="expressions/">expressions</a></td>
<td>HGVS, SPDI, and gnomAD-style strings should be represented as Expressions</td>
</tr>
<tr>
<td><a href="extensions/">extensions</a></td>
<td></td>
</tr>
<tr>
<td><a href="externalReference/">externalReference</a></td>
<td></td>
</tr>
<tr>
<td><a href="externalReferences/">externalReferences</a></td>
<td>External identifiers for this message</td>
</tr>
<tr>
<td><a href="familyId/">familyId</a></td>
<td></td>
</tr>
<tr>
<td><a href="fileAttributes/">fileAttributes</a></td>
<td>Map of attributes describing the file</td>
</tr>
<tr>
<td><a href="files/">files</a></td>
<td>Pointer to relevant file(s) for the cohort</td>
</tr>
<tr>
<td><a href="filter/">filter</a></td>
<td></td>
</tr>
<tr>
<td><a href="fractions/">fractions</a></td>
<td>The total number of fractions delivered as part of treatment</td>
</tr>
<tr>
<td><a href="gender/">gender</a></td>
<td>Self-identified gender</td>
</tr>
<tr>
<td><a href="gene/">gene</a></td>
<td></td>
</tr>
<tr>
<td><a href="geneContext/">geneContext</a></td>
<td>A specific gene context that applies to this variant</td>
</tr>
<tr>
<td><a href="geneId/">geneId</a></td>
<td></td>
</tr>
<tr>
<td><a href="genomeAssembly/">genomeAssembly</a></td>
<td></td>
</tr>
<tr>
<td><a href="genomicInterpretations/">genomicInterpretations</a></td>
<td>genomic features containing the status of their contribution towards the diag...</td>
</tr>
<tr>
<td><a href="gestationalAge/">gestationalAge</a></td>
<td></td>
</tr>
<tr>
<td><a href="haplotype/">haplotype</a></td>
<td></td>
</tr>
<tr>
<td><a href="high/">high</a></td>
<td></td>
</tr>
<tr>
<td><a href="histologicalDiagnosis/">histologicalDiagnosis</a></td>
<td>This is the pathologist’s diagnosis and may often represent a refinement of t...</td>
</tr>
<tr>
<td><a href="id/">id</a></td>
<td></td>
</tr>
<tr>
<td><a href="indefiniteRange/">indefiniteRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="individualId/">individualId</a></td>
<td></td>
</tr>
<tr>
<td><a href="individualToFileIdentifiers/">individualToFileIdentifiers</a></td>
<td>A map of identifiers mapping an individual to a sample in the file</td>
</tr>
<tr>
<td><a href="info/">info</a></td>
<td></td>
</tr>
<tr>
<td><a href="interpretations/">interpretations</a></td>
<td></td>
</tr>
<tr>
<td><a href="interpretationStatus/">interpretationStatus</a></td>
<td></td>
</tr>
<tr>
<td><a href="interval/">interval</a></td>
<td></td>
</tr>
<tr>
<td><a href="iriPrefix/">iriPrefix</a></td>
<td>Full IRI prefix which can be used with the namespace_prefix and the OntologyC...</td>
</tr>
<tr>
<td><a href="iso8601duration/">iso8601duration</a></td>
<td>The :ref:<code>ISO 8601&lt;metadata_date_time&gt;</code> age of this object as ISO8601 duratio...</td>
</tr>
<tr>
<td><a href="karyotypicSex/">karyotypicSex</a></td>
<td>The karyotypic sex of the individual</td>
</tr>
<tr>
<td><a href="label/">label</a></td>
<td>class label, aka name</td>
</tr>
<tr>
<td><a href="laterality/">laterality</a></td>
<td>The term used to indicate laterality of diagnosis, if applicable</td>
</tr>
<tr>
<td><a href="literalSequenceExpression/">literalSequenceExpression</a></td>
<td></td>
</tr>
<tr>
<td><a href="location/">location</a></td>
<td></td>
</tr>
<tr>
<td><a href="low/">low</a></td>
<td></td>
</tr>
<tr>
<td><a href="materialSample/">materialSample</a></td>
<td>This element can be used to specify the status of the sample</td>
</tr>
<tr>
<td><a href="maternalId/">maternalId</a></td>
<td></td>
</tr>
<tr>
<td><a href="max/">max</a></td>
<td></td>
</tr>
<tr>
<td><a href="measurements/">measurements</a></td>
<td>Quantifiable measurements related to the individual</td>
</tr>
<tr>
<td><a href="medicalActions/">medicalActions</a></td>
<td></td>
</tr>
<tr>
<td><a href="members/">members</a></td>
<td></td>
</tr>
<tr>
<td><a href="metaData/">metaData</a></td>
<td>Structured definitions of the resources and ontologies used within the phenop...</td>
</tr>
<tr>
<td><a href="min/">min</a></td>
<td></td>
</tr>
<tr>
<td><a href="modality/">modality</a></td>
<td>The modality of radiation therapy (e</td>
</tr>
<tr>
<td><a href="modifiers/">modifiers</a></td>
<td>subclasses of HP:0012823 ! Clinical modifier apart from Severity HP:0012824 -...</td>
</tr>
<tr>
<td><a href="moleculeContext/">moleculeContext</a></td>
<td>The molecular context of the vrs variation</td>
</tr>
<tr>
<td><a href="name/">name</a></td>
<td>e</td>
</tr>
<tr>
<td><a href="namespacePrefix/">namespacePrefix</a></td>
<td>The prefix used in the CURIE of an OntologyClass e</td>
</tr>
<tr>
<td><a href="nanos/">nanos</a></td>
<td>Non-negative fractions of a second at nanosecond resolution</td>
</tr>
<tr>
<td><a href="number/">number</a></td>
<td></td>
</tr>
<tr>
<td><a href="onset/">onset</a></td>
<td>the values of this will come from the HPO onset hierarchy i</td>
</tr>
<tr>
<td><a href="ontologyClass/">ontologyClass</a></td>
<td></td>
</tr>
<tr>
<td><a href="paternalId/">paternalId</a></td>
<td></td>
</tr>
<tr>
<td><a href="pathologicalStage/">pathologicalStage</a></td>
<td>ARGO mapping specimen::pathological_tumour_staging_system ARGO mapping specim...</td>
</tr>
<tr>
<td><a href="pathologicalTnmFinding/">pathologicalTnmFinding</a></td>
<td>ARGO mapping specimen::pathological_t_category ARGO mapping specimen::patholo...</td>
</tr>
<tr>
<td><a href="pedigree/">pedigree</a></td>
<td>The pedigree defining the relations between the proband and their relatives</td>
</tr>
<tr>
<td><a href="performed/">performed</a></td>
<td>When the procedure was performed</td>
</tr>
<tr>
<td><a href="persons/">persons</a></td>
<td></td>
</tr>
<tr>
<td><a href="phenopacketSchemaVersion/">phenopacketSchemaVersion</a></td>
<td>phenopacket-schema-version used to create this phenopacket</td>
</tr>
<tr>
<td><a href="phenotypicFeatures/">phenotypicFeatures</a></td>
<td>Phenotypic features relating to the subject of the phenopacket</td>
</tr>
<tr>
<td><a href="pos/">pos</a></td>
<td></td>
</tr>
<tr>
<td><a href="primarySite/">primarySite</a></td>
<td>The text term used to describe the primary site of disease, as categorized by...</td>
</tr>
<tr>
<td><a href="proband/">proband</a></td>
<td>The individual representing the focus of this packet - e</td>
</tr>
<tr>
<td><a href="procedure/">procedure</a></td>
<td></td>
</tr>
<tr>
<td><a href="progressStatus/">progressStatus</a></td>
<td></td>
</tr>
<tr>
<td><a href="qual/">qual</a></td>
<td></td>
</tr>
<tr>
<td><a href="quantity/">quantity</a></td>
<td></td>
</tr>
<tr>
<td><a href="radiationTherapy/">radiationTherapy</a></td>
<td></td>
</tr>
<tr>
<td><a href="ref/">ref</a></td>
<td></td>
</tr>
<tr>
<td><a href="reference/">reference</a></td>
<td>FHIR mapping: Condition</td>
</tr>
<tr>
<td><a href="referenceRange/">referenceRange</a></td>
<td>Reference range for the quantity e</td>
</tr>
<tr>
<td><a href="regimenStatus/">regimenStatus</a></td>
<td></td>
</tr>
<tr>
<td><a href="relatives/">relatives</a></td>
<td>Individuals related in some way to the patient</td>
</tr>
<tr>
<td><a href="repeatedSequenceExpression/">repeatedSequenceExpression</a></td>
<td></td>
</tr>
<tr>
<td><a href="resolution/">resolution</a></td>
<td></td>
</tr>
<tr>
<td><a href="resources/">resources</a></td>
<td>a listing of the ontologies and resources referenced in the phenopacket</td>
</tr>
<tr>
<td><a href="responseToTreatment/">responseToTreatment</a></td>
<td>ARGO mapping treatment::response_to_treatment</td>
</tr>
<tr>
<td><a href="reverseComplement/">reverseComplement</a></td>
<td></td>
</tr>
<tr>
<td><a href="routeOfAdministration/">routeOfAdministration</a></td>
<td></td>
</tr>
<tr>
<td><a href="sampledTissue/">sampledTissue</a></td>
<td>UBERON class describing the tissue from which the specimen was collected</td>
</tr>
<tr>
<td><a href="sampleProcessing/">sampleProcessing</a></td>
<td>Field to represent how the sample was processed</td>
</tr>
<tr>
<td><a href="sampleStorage/">sampleStorage</a></td>
<td>Field to represent how the sample was stored ARGO mapping specimen::specimen_...</td>
</tr>
<tr>
<td><a href="sampleType/">sampleType</a></td>
<td>Recommended use of EFO term to describe the sample</td>
</tr>
<tr>
<td><a href="scheduleFrequency/">scheduleFrequency</a></td>
<td></td>
</tr>
<tr>
<td><a href="seconds/">seconds</a></td>
<td>Represents seconds of UTC time since Unix epoch 1970-01-01T00:00:00Z</td>
</tr>
<tr>
<td><a href="sequence/">sequence</a></td>
<td></td>
</tr>
<tr>
<td><a href="sequenceId/">sequenceId</a></td>
<td></td>
</tr>
<tr>
<td><a href="sequenceInterval/">sequenceInterval</a></td>
<td></td>
</tr>
<tr>
<td><a href="sequenceLocation/">sequenceLocation</a></td>
<td></td>
</tr>
<tr>
<td><a href="severity/">severity</a></td>
<td>Severity of the condition e</td>
</tr>
<tr>
<td><a href="sex/">sex</a></td>
<td></td>
</tr>
<tr>
<td><a href="speciesId/">speciesId</a></td>
<td></td>
</tr>
<tr>
<td><a href="start/">start</a></td>
<td></td>
</tr>
<tr>
<td><a href="startDefiniteRange/">startDefiniteRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="startIndefiniteRange/">startIndefiniteRange</a></td>
<td></td>
</tr>
<tr>
<td><a href="startNumber/">startNumber</a></td>
<td></td>
</tr>
<tr>
<td><a href="startTime/">startTime</a></td>
<td>possibly undefined;</td>
</tr>
<tr>
<td><a href="status/">status</a></td>
<td></td>
</tr>
<tr>
<td><a href="structuralType/">structuralType</a></td>
<td>The structural variant type associated with this variant, such as a substitut...</td>
</tr>
<tr>
<td><a href="subject/">subject</a></td>
<td>The individual representing the focus of this packet - e</td>
</tr>
<tr>
<td><a href="subjectOrBiosampleId/">subjectOrBiosampleId</a></td>
<td>identifier for the subject of the interpretation</td>
</tr>
<tr>
<td><a href="submittedBy/">submittedBy</a></td>
<td>information about the person/organisation/network that has submitted this phe...</td>
</tr>
<tr>
<td><a href="summary/">summary</a></td>
<td></td>
</tr>
<tr>
<td><a href="survivalTimeInDays/">survivalTimeInDays</a></td>
<td>ARGO mapping donor::survival_time</td>
</tr>
<tr>
<td><a href="symbol/">symbol</a></td>
<td>The primary gene symbol</td>
</tr>
<tr>
<td><a href="syntax/">syntax</a></td>
<td></td>
</tr>
<tr>
<td><a href="taxonomy/">taxonomy</a></td>
<td>NCBI taxonomic identifier NCBITaxon e</td>
</tr>
<tr>
<td><a href="term/">term</a></td>
<td>The identifier of this disease e</td>
</tr>
<tr>
<td><a href="text/">text</a></td>
<td></td>
</tr>
<tr>
<td><a href="therapeuticActionability/">therapeuticActionability</a></td>
<td></td>
</tr>
<tr>
<td><a href="therapeuticRegimen/">therapeuticRegimen</a></td>
<td></td>
</tr>
<tr>
<td><a href="timeAtLastEncounter/">timeAtLastEncounter</a></td>
<td>An TimeElement object describing the age of the individual at the last time o...</td>
</tr>
<tr>
<td><a href="timeObserved/">timeObserved</a></td>
<td>The time at which the measurement was made</td>
</tr>
<tr>
<td><a href="timeOfCollection/">timeOfCollection</a></td>
<td>An TimeElement describing either the age of the individual this biosample was...</td>
</tr>
<tr>
<td><a href="timeOfDeath/">timeOfDeath</a></td>
<td></td>
</tr>
<tr>
<td><a href="timestamp/">timestamp</a></td>
<td></td>
</tr>
<tr>
<td><a href="treatment/">treatment</a></td>
<td></td>
</tr>
<tr>
<td><a href="treatmentIntent/">treatmentIntent</a></td>
<td>Whether the intention of the treatment was curative, palliative, ARGO mapping...</td>
</tr>
<tr>
<td><a href="treatmentTarget/">treatmentTarget</a></td>
<td>The condition or disease that this treatment was intended to address</td>
</tr>
<tr>
<td><a href="treatmentTerminationReason/">treatmentTerminationReason</a></td>
<td>ARGO mapping treatment::treatment_outcome</td>
</tr>
<tr>
<td><a href="tumorGrade/">tumorGrade</a></td>
<td>Potentially a child term of NCIT:C28076 (Disease Grade Qualifier) or equivale...</td>
</tr>
<tr>
<td><a href="tumorProgression/">tumorProgression</a></td>
<td>Is the specimen tissue from the primary tumor, a metastasis or a recurrence? ...</td>
</tr>
<tr>
<td><a href="type/">type</a></td>
<td>The primary ontology class which describes the phenotype</td>
</tr>
<tr>
<td><a href="typedQuantities/">typedQuantities</a></td>
<td>The quantities required to fully describe the complex value</td>
</tr>
<tr>
<td><a href="typeUrl/">typeUrl</a></td>
<td>A URL/resource name that uniquely identifies the type of the serialized proto...</td>
</tr>
<tr>
<td><a href="unit/">unit</a></td>
<td>For instance, NCIT subhierarchy, Unit of Measure (Code C25709), https://www</td>
</tr>
<tr>
<td><a href="updatedBy/">updatedBy</a></td>
<td>Information about the person/organisation/network that has updated the phenop...</td>
</tr>
<tr>
<td><a href="updates/">updates</a></td>
<td>An OPTIONAL list of Updates to the phenopacket</td>
</tr>
<tr>
<td><a href="uri/">uri</a></td>
<td>URI for the file e</td>
</tr>
<tr>
<td><a href="url/">url</a></td>
<td>For OBO ontologies, this should always be the PURL, e</td>
</tr>
<tr>
<td><a href="value/">value</a></td>
<td></td>
</tr>
<tr>
<td><a href="valueId/">valueId</a></td>
<td>The official gene identifier as designated by the organism gene nomenclature ...</td>
</tr>
<tr>
<td><a href="variantInterpretation/">variantInterpretation</a></td>
<td></td>
</tr>
<tr>
<td><a href="variation/">variation</a></td>
<td></td>
</tr>
<tr>
<td><a href="variationDescriptor/">variationDescriptor</a></td>
<td></td>
</tr>
<tr>
<td><a href="variationSet/">variationSet</a></td>
<td></td>
</tr>
<tr>
<td><a href="vcfRecord/">vcfRecord</a></td>
<td>A VCF Record of the variant</td>
</tr>
<tr>
<td><a href="version/">version</a></td>
<td>for OBO ontologies, this should be the versionIRI</td>
</tr>
<tr>
<td><a href="vitalStatus/">vitalStatus</a></td>
<td>Vital status of the individual</td>
</tr>
<tr>
<td><a href="vrsRefAlleleSeq/">vrsRefAlleleSeq</a></td>
<td>A Sequence corresponding to a “ref allele”, describing the sequence expected ...</td>
</tr>
<tr>
<td><a href="weeks/">weeks</a></td>
<td></td>
</tr>
<tr>
<td><a href="xrefs/">xrefs</a></td>
<td>Related concept IDs (e</td>
</tr>
</tbody>
</table>
<h2 id="enumerations">Enumerations</h2>
<table>
<thead>
<tr>
<th>Enumeration</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="AcmgPathogenicityClassification/">AcmgPathogenicityClassification</a></td>
<td></td>
</tr>
<tr>
<td><a href="AffectedStatus/">AffectedStatus</a></td>
<td></td>
</tr>
<tr>
<td><a href="AllelicStateTerms/">AllelicStateTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="AssaysTerms/">AssaysTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="DrugType/">DrugType</a></td>
<td>A simplified version of ODHSI-DRUG_EXPOSURE</td>
</tr>
<tr>
<td><a href="GenderTerms/">GenderTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="GENOZygosityTerms/">GENOZygosityTerms</a></td>
<td>GENO is an ontology of genotypes their more fundamental sequence components</td>
</tr>
<tr>
<td><a href="HGNCGeneTerms/">HGNCGeneTerms</a></td>
<td>The HUGO Gene Nomenclature Committee (HGNC) provides standard names, symbols,...</td>
</tr>
<tr>
<td><a href="HPOAbnormalityTerms/">HPOAbnormalityTerms</a></td>
<td>The Human Phenotype Ontology (HPO) provides a comprehensive logical standard ...</td>
</tr>
<tr>
<td><a href="InterpretationStatus/">InterpretationStatus</a></td>
<td></td>
</tr>
<tr>
<td><a href="KaryotypicSex/">KaryotypicSex</a></td>
<td>Karyotypic sex of the individual</td>
</tr>
<tr>
<td><a href="LateralityTerms/">LateralityTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="LOINCMeasurementTerms/">LOINCMeasurementTerms</a></td>
<td>Logical Observation Identifiers Names and Codes (LOINC) is a database and uni...</td>
</tr>
<tr>
<td><a href="MedicalActionsTerms/">MedicalActionsTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="MoleculeContext/">MoleculeContext</a></td>
<td></td>
</tr>
<tr>
<td><a href="MondoDiseaseTerms/">MondoDiseaseTerms</a></td>
<td>Mondo Disease Ontology provides a comprehensive logically structured ontology...</td>
</tr>
<tr>
<td><a href="NCITDiseaseTerms/">NCITDiseaseTerms</a></td>
<td>All disease terms from the NCI Thesaurus</td>
</tr>
<tr>
<td><a href="NCITNeoplasmTerms/">NCITNeoplasmTerms</a></td>
<td>All neoplasm terms from the NCI Thesaurus</td>
</tr>
<tr>
<td><a href="OnsetTerms/">OnsetTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="OrganTerms/">OrganTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="ProgressStatus/">ProgressStatus</a></td>
<td></td>
</tr>
<tr>
<td><a href="RegimenStatus/">RegimenStatus</a></td>
<td></td>
</tr>
<tr>
<td><a href="ResponseTerms/">ResponseTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="Sex/">Sex</a></td>
<td>Sex of an individual FHIR mapping: AdministrativeGender (https://www</td>
</tr>
<tr>
<td><a href="SpatialPatternTerms/">SpatialPatternTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="Status/">Status</a></td>
<td>Default = false i</td>
</tr>
<tr>
<td><a href="TherapeuticActionability/">TherapeuticActionability</a></td>
<td></td>
</tr>
<tr>
<td><a href="UberonAnatomicalEntityTerms/">UberonAnatomicalEntityTerms</a></td>
<td>UBERON is an integrated cross-species ontology with classes representing a va...</td>
</tr>
<tr>
<td><a href="UnitTerms/">UnitTerms</a></td>
<td></td>
</tr>
<tr>
<td><a href="UOUnitTerms/">UOUnitTerms</a></td>
<td>The Units of measurement ontology (denoted UO) provides terms for units commo...</td>
</tr>
</tbody>
</table>
<h2 id="types">Types</h2>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="xsd:boolean">xsd:boolean</a></td>
<td>A binary (true or false) value</td>
</tr>
<tr>
<td><a href="xsd:date">xsd:date</a></td>
<td>a date (year, month and day) in an idealized calendar</td>
</tr>
<tr>
<td><a href="https://w3id.org/linkml/DateOrDatetime">linkml:DateOrDatetime</a></td>
<td>Either a date or a datetime</td>
</tr>
<tr>
<td><a href="xsd:dateTime">xsd:dateTime</a></td>
<td>The combination of a date and time</td>
</tr>
<tr>
<td><a href="xsd:decimal">xsd:decimal</a></td>
<td>A real number with arbitrary precision that conforms to the xsd:decimal speci...</td>
</tr>
<tr>
<td><a href="xsd:double">xsd:double</a></td>
<td>A real number that conforms to the xsd:double specification</td>
</tr>
<tr>
<td><a href="xsd:float">xsd:float</a></td>
<td>A real number that conforms to the xsd:float specification</td>
</tr>
<tr>
<td><a href="xsd:integer">xsd:integer</a></td>
<td>An integer</td>
</tr>
<tr>
<td><a href="xsd:string">xsd:string</a></td>
<td>Prefix part of CURIE</td>
</tr>
<tr>
<td><a href="shex:nonLiteral">shex:nonLiteral</a></td>
<td>A URI, CURIE or BNODE that represents a node in a model</td>
</tr>
<tr>
<td><a href="shex:iri">shex:iri</a></td>
<td>A URI or CURIE that represents an object in the model</td>
</tr>
<tr>
<td><a href="xsd:string">xsd:string</a></td>
<td>A character string</td>
</tr>
<tr>
<td><a href="xsd:dateTime">xsd:dateTime</a></td>
<td>A time object represents a (local) time of day, independent of any particular...</td>
</tr>
<tr>
<td><a href="xsd:anyURI">xsd:anyURI</a></td>
<td>a complete URI</td>
</tr>
<tr>
<td><a href="xsd:anyURI">xsd:anyURI</a></td>
<td>a URI or a CURIE</td>
</tr>
</tbody>
</table>
<h2 id="subsets">Subsets</h2>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</article>
</div>
<script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>
</div>
</main>
<footer class="md-footer">
<nav aria-label="Footer" class="md-footer__inner md-grid">
<a aria-label="Previous: Home" class="md-footer__link md-footer__link--prev" href="home/" rel="prev">
<div class="md-footer__button md-icon">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</div>
<div class="md-footer__title">
<div class="md-ellipsis">
<span class="md-footer__direction">
                Previous
              </span>
              Home
            </div>
</div>
</a>
</nav>
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": ".", "features": ["content.tabs.link"], "search": "assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
<script src="assets/javascripts/bundle.d6c3db9e.min.js"></script>
<script src="https://unpkg.com/mermaid@9.3.0/dist/mermaid.min.js"></script><script>mermaid.initialize({});</script></body>
</html>