
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="https://cmungall.github.io/linkml-phenopackets/Biosample/" rel="canonical"/>
<link href="../assets/images/favicon.png" rel="icon"/>
<meta content="mkdocs-1.6.0, mkdocs-material-8.5.10" name="generator"/>
<title>Class: Biosample - phenopackets in linkml EXPERIMENTAL DEMO</title>
<link href="../assets/stylesheets/main.975780f9.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.2505c338.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<link href="../css/ansi-colours.css" rel="stylesheet"/>
<link href="../css/jupyter-cells.css" rel="stylesheet"/>
<link href="../css/pandas-dataframe.css" rel="stylesheet"/>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
</head>
<body data-md-color-accent="amber" data-md-color-primary="teal" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#class-biosample">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="phenopackets in linkml EXPERIMENTAL DEMO" class="md-header__button md-logo" data-md-component="logo" href=".." title="phenopackets in linkml EXPERIMENTAL DEMO">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            phenopackets in linkml EXPERIMENTAL DEMO
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Class: Biosample
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="amber" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="teal" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 2a7 7 0 0 0-7 7c0 2.38 1.19 4.47 3 5.74V17a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1v-2.26c1.81-1.27 3-3.36 3-5.74a7 7 0 0 0-7-7M9 21a1 1 0 0 0 1 1h4a1 1 0 0 0 1-1v-1H9v1Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="amber" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="teal" data-md-color-scheme="slate" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 2a7 7 0 0 1 7 7c0 2.38-1.19 4.47-3 5.74V17a1 1 0 0 1-1 1H9a1 1 0 0 1-1-1v-2.26C6.19 13.47 5 11.38 5 9a7 7 0 0 1 7-7M9 21v-1h6v1a1 1 0 0 1-1 1h-4a1 1 0 0 1-1-1m3-17a5 5 0 0 0-5 5c0 2.05 1.23 3.81 3 4.58V16h4v-2.42c1.77-.77 3-2.53 3-4.58a5 5 0 0 0-5-5Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/cmungall/linkml-phenopackets/" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="phenopackets in linkml EXPERIMENTAL DEMO" class="md-nav__button md-logo" data-md-component="logo" href=".." title="phenopackets in linkml EXPERIMENTAL DEMO">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"></path></svg>
</a>
    phenopackets in linkml EXPERIMENTAL DEMO
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/cmungall/linkml-phenopackets/" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../home/">
        Home
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="..">
        Index
      </a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
</label>
<nav aria-label="Notebooks" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../notebooks/LinkML-Validate/">
        Validating
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../notebooks/Pydantic/">
        Pydantic
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../notebooks/OAK/">
        Using OAK
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../notebooks/Generating-RDF/">
        RDF
      </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#slots">
    Slots
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usages">
    Usages
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#identifier-and-mapping-information">
    Identifier and Mapping Information
  </a>
<nav aria-label="Identifier and Mapping Information" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#schema-source">
    Schema Source
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#mappings">
    Mappings
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#linkml-source">
    LinkML Source
  </a>
<nav aria-label="LinkML Source" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#direct">
    Direct
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#induced">
    Induced
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<a class="md-content__button md-icon" href="https://github.com/cmungall/linkml-phenopackets/edit/master/docs/Biosample.md" title="Edit this page">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg>
</a>
<h1 id="class-biosample">Class: Biosample</h1>
<p><em>A Biosample refers to a unit of biological material from which the substrate molecules (e.g. genomic DNA, RNA, proteins) for molecular analyses (e.g. sequencing, array hybridisation, mass-spectrometry) are extracted. Examples would be a tissue biopsy, a single cell from a culture for single cell genome sequencing or a protein fraction from a gradient centrifugation. Several instances (e.g. technical replicates) or types of experiments (e.g. genomic array as well as RNA-seq experiments) may refer to the same Biosample. FHIR mapping: Specimen (http://www.hl7.org/fhir/specimen.html).</em></p>
<p>URI: <a href="https://w3id.org/linkml/phenopackets/phenopackets/Biosample">phenopackets:Biosample</a></p>
<div class="mermaid">erDiagram
Biosample {
    string derivedFromId  
    string description  
    string id  
    string individualId  
}
OntologyClass {
    string id  
    string label  
}
TimeElement {
    string timestamp  
}
TimeInterval {
    string end  
    string start  
}
GestationalAge {
    integer days  
    integer weeks  
}
AgeRange {

}
Age {
    string iso8601duration  
}
Procedure {

}
PhenotypicFeature {
    string description  
    boolean excluded  
}
Evidence {

}
Measurement {
    string description  
}
Value {

}
ComplexValue {

}
File {
    string uri  
}
Dictionary {

}

Biosample ||--}o OntologyClass : "diagnosticMarkers"
Biosample ||--}o File : "files"
Biosample ||--|o OntologyClass : "histologicalDiagnosis"
Biosample ||--|o OntologyClass : "materialSample"
Biosample ||--}o Measurement : "measurements"
Biosample ||--|o OntologyClass : "pathologicalStage"
Biosample ||--}o OntologyClass : "pathologicalTnmFinding"
Biosample ||--}o PhenotypicFeature : "phenotypicFeatures"
Biosample ||--|o Procedure : "procedure"
Biosample ||--|o OntologyClass : "sampleProcessing"
Biosample ||--|o OntologyClass : "sampleStorage"
Biosample ||--|o OntologyClass : "sampleType"
Biosample ||--|o OntologyClass : "sampledTissue"
Biosample ||--|o OntologyClass : "taxonomy"
Biosample ||--|o TimeElement : "timeOfCollection"
Biosample ||--|o OntologyClass : "tumorGrade"
Biosample ||--|o OntologyClass : "tumorProgression"
TimeElement ||--|o Age : "age"
TimeElement ||--|o AgeRange : "ageRange"
TimeElement ||--|o GestationalAge : "gestationalAge"
TimeElement ||--|o TimeInterval : "interval"
TimeElement ||--|o OntologyClass : "ontologyClass"
AgeRange ||--|o Age : "end"
AgeRange ||--|o Age : "start"
Procedure ||--|o OntologyClass : "bodySite"
Procedure ||--|o OntologyClass : "code"
Procedure ||--|o TimeElement : "performed"
PhenotypicFeature ||--}o Evidence : "evidence"
PhenotypicFeature ||--}o OntologyClass : "modifiers"
PhenotypicFeature ||--|o TimeElement : "onset"
PhenotypicFeature ||--|o TimeElement : "resolution"
PhenotypicFeature ||--|o OntologyClass : "severity"
PhenotypicFeature ||--|o OntologyClass : "type"
Evidence ||--|o OntologyClass : "evidenceCode"
Evidence ||--|o ExternalReference : "reference"
Measurement ||--|o OntologyClass : "assay"
Measurement ||--|o ComplexValue : "complexValue"
Measurement ||--|o Procedure : "procedure"
Measurement ||--|o TimeElement : "timeObserved"
Measurement ||--|o Value : "value"
Value ||--|o OntologyClass : "ontologyClass"
Value ||--|o Quantity : "quantity"
ComplexValue ||--}o TypedQuantity : "typedQuantities"
File ||--|o Dictionary : "fileAttributes"
File ||--|o Dictionary : "individualToFileIdentifiers"

</div>
<!-- no inheritance hierarchy -->
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Cardinality and Range</th>
<th>Description</th>
<th>Inheritance</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../derivedFromId/">derivedFromId</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>The id of the parent biosample this biosample was derived from</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../description/">description</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>The biosample's description</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../diagnosticMarkers/">diagnosticMarkers</a></td>
<td>0..* <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>Clinically relevant bio markers</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../files/">files</a></td>
<td>0..* <br/> <a href="../File/">File</a></td>
<td>Pointer to the relevant file(s) for the biosample</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../histologicalDiagnosis/">histologicalDiagnosis</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>This is the pathologist’s diagnosis and may often represent a refinement of t...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../id/">id</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>biosamples SAMN08666232 Human Cell Atlas The Biosample id This is unique in t...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../individualId/">individualId</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>The id of the individual this biosample was derived from</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../materialSample/">materialSample</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>This element can be used to specify the status of the sample</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../measurements/">measurements</a></td>
<td>0..* <br/> <a href="../Measurement/">Measurement</a></td>
<td></td>
<td>direct</td>
</tr>
<tr>
<td><a href="../pathologicalStage/">pathologicalStage</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>ARGO mapping specimen::pathological_tumour_staging_system ARGO mapping specim...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../pathologicalTnmFinding/">pathologicalTnmFinding</a></td>
<td>0..* <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>ARGO mapping specimen::pathological_t_category ARGO mapping specimen::patholo...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../phenotypicFeatures/">phenotypicFeatures</a></td>
<td>0..* <br/> <a href="../PhenotypicFeature/">PhenotypicFeature</a></td>
<td>Phenotypic characteristics of the BioSample, for example histological finding...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../procedure/">procedure</a></td>
<td>0..1 <br/> <a href="../Procedure/">Procedure</a></td>
<td>Clinical procedure performed on the subject in order to extract the biosample</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../sampleProcessing/">sampleProcessing</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>Field to represent how the sample was processed</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../sampleStorage/">sampleStorage</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>Field to represent how the sample was stored ARGO mapping specimen::specimen_...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../sampleType/">sampleType</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>Recommended use of EFO term to describe the sample</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../sampledTissue/">sampledTissue</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>UBERON class describing the tissue from which the specimen was collected</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../taxonomy/">taxonomy</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>NCBI taxonomic identifier (NCBITaxon) of the sample e</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../timeOfCollection/">timeOfCollection</a></td>
<td>0..1 <br/> <a href="../TimeElement/">TimeElement</a></td>
<td>An TimeElement describing either the age of the individual this biosample was...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../tumorGrade/">tumorGrade</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>Potentially a child term of NCIT:C28076 (Disease Grade Qualifier) or equivale...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../tumorProgression/">tumorProgression</a></td>
<td>0..1 <br/> <a href="../OntologyClass/">OntologyClass</a></td>
<td>Is the specimen tissue from the primary tumor, a metastasis or a recurrence? ...</td>
<td>direct</td>
</tr>
</tbody>
</table>
<h2 id="usages">Usages</h2>
<table>
<thead>
<tr>
<th>used by</th>
<th>used in</th>
<th>type</th>
<th>used</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../Phenopacket/">Phenopacket</a></td>
<td><a href="../biosamples/">biosamples</a></td>
<td>range</td>
<td><a href="./">Biosample</a></td>
</tr>
</tbody>
</table>
<h2 id="identifier-and-mapping-information">Identifier and Mapping Information</h2>
<h3 id="schema-source">Schema Source</h3>
<ul>
<li>from schema: https://w3id.org/linkml/phenopackets/phenopackets</li>
</ul>
<h2 id="mappings">Mappings</h2>
<table>
<thead>
<tr>
<th>Mapping Type</th>
<th>Mapped Value</th>
</tr>
</thead>
<tbody>
<tr>
<td>self</td>
<td>phenopackets:Biosample</td>
</tr>
<tr>
<td>native</td>
<td>phenopackets:Biosample</td>
</tr>
</tbody>
</table>
<h2 id="linkml-source">LinkML Source</h2>
<!-- TODO: investigate https://stackoverflow.com/questions/37606292/how-to-create-tabbed-code-blocks-in-mkdocs-or-sphinx -->
<h3 id="direct">Direct</h3>
<details>
<pre><code class="language-yaml">name: Biosample
description: 'A Biosample refers to a unit of biological material from which the substrate
  molecules (e.g. genomic DNA, RNA, proteins) for molecular analyses (e.g. sequencing,
  array hybridisation, mass-spectrometry) are extracted. Examples would be a tissue
  biopsy, a single cell from a culture for single cell genome sequencing or a protein
  fraction from a gradient centrifugation. Several instances (e.g. technical replicates)
  or types of experiments (e.g. genomic array as well as RNA-seq experiments) may
  refer to the same Biosample. FHIR mapping: Specimen (http://www.hl7.org/fhir/specimen.html).'
from_schema: https://w3id.org/linkml/phenopackets/phenopackets
attributes:
  derivedFromId:
    name: derivedFromId
    annotations:
      rank:
        tag: rank
        value: 3
    description: The id of the parent biosample this biosample was derived from.
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    domain_of:
    - Biosample
    range: string
  description:
    name: description
    annotations:
      rank:
        tag: rank
        value: 4
    description: The biosample's description. This attribute contains human readable
      text. The "description" attributes should not contain any structured data.
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    domain_of:
    - Cohort
    - ExternalReference
    - Measurement
    - PhenotypicFeature
    - Biosample
    - GeneDescriptor
    - VariationDescriptor
    range: string
  diagnosticMarkers:
    name: diagnosticMarkers
    annotations:
      rank:
        tag: rank
        value: 16
    description: Clinically relevant bio markers. Most of the assays such as IHC are
      covered by the NCIT under the sub-hierarchy NCIT:C25294 (Laboratory Procedure).
      e.g. NCIT:C68748 (HER2/Neu Positive), NCIT:C131711 (Human Papillomavirus-18
      Positive)
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    multivalued: true
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  files:
    name: files
    annotations:
      rank:
        tag: rank
        value: 18
    description: Pointer to the relevant file(s) for the biosample. Files relating
      to the entire individual e.g. a germline exome/genome should be associated with
      the Phenopacket rather than the Biosample it was derived from.
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    multivalued: true
    domain_of:
    - Cohort
    - Family
    - Phenopacket
    - Biosample
    range: File
    inlined: true
    inlined_as_list: true
  histologicalDiagnosis:
    name: histologicalDiagnosis
    annotations:
      rank:
        tag: rank
        value: 11
    description: This is the pathologist’s diagnosis and may often represent a refinement
      of the clinical diagnosis given in the Patient/Clinical module. Should use the
      same terminology as diagnosis, but represent the pathologist’s findings. Normal
      samples would be tagged with the term "NCIT:C38757", "Negative Finding" ARGO
      mapping specimen::tumour_histological_type
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.tumour_histological_type
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  id:
    name: id
    annotations:
      percent_encoded:
        tag: percent_encoded
        value: true
      rank:
        tag: rank
        value: 1
    description: biosamples SAMN08666232 Human Cell Atlas The Biosample id This is
      unique in the context of the server instance. ARGO mapping specimen::submitter_specimen_id
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.submitter_specimen_id
    domain_of:
    - Cohort
    - Family
    - Phenopacket
    - ExternalReference
    - OntologyClass
    - Biosample
    - Allele
    - ChromosomeLocation
    - CopyNumber
    - Member
    - SequenceLocation
    - Text
    - VariationDescriptor
    - VcfRecord
    - Interpretation
    - Individual
    - Resource
    range: string
  individualId:
    name: individualId
    annotations:
      rank:
        tag: rank
        value: 2
    description: The id of the individual this biosample was derived from. ARGO mapping
      specimen::submitter_donor_id
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.submitter_donor_id
    rank: 1000
    domain_of:
    - Biosample
    - Person
    range: string
  materialSample:
    name: materialSample
    annotations:
      rank:
        tag: rank
        value: 19
    description: 'This element can be used to specify the status of the sample. For
      instance, a status may be used as a normal control, often in combination with
      another sample that is thought to contain a pathological finding. We recommend
      use of ontology terms such as: EFO:0009654 (reference sample) or EFO:0009655
      (abnormal sample) ARGO mapping sample_registration::tumour_normal_designation'
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:sample_registration.tumour_normal_designation
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  measurements:
    name: measurements
    annotations:
      rank:
        tag: rank
        value: 8
    description: ''
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    multivalued: true
    domain_of:
    - Phenopacket
    - Biosample
    range: Measurement
    inlined: true
    inlined_as_list: true
  pathologicalStage:
    name: pathologicalStage
    annotations:
      rank:
        tag: rank
        value: 14
    description: ARGO mapping specimen::pathological_tumour_staging_system ARGO mapping
      specimen::pathological_stage_group
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.pathological_tumour_staging_system
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  pathologicalTnmFinding:
    name: pathologicalTnmFinding
    annotations:
      rank:
        tag: rank
        value: 15
    description: ARGO mapping specimen::pathological_t_category ARGO mapping specimen::pathological_n_category
      ARGO mapping specimen::pathological_m_category
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.pathological_t_category
    rank: 1000
    multivalued: true
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  phenotypicFeatures:
    name: phenotypicFeatures
    annotations:
      rank:
        tag: rank
        value: 7
    description: Phenotypic characteristics of the BioSample, for example histological
      findings of a biopsy.
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    multivalued: true
    domain_of:
    - Phenopacket
    - Biosample
    range: PhenotypicFeature
    inlined: true
    inlined_as_list: true
  procedure:
    name: procedure
    annotations:
      rank:
        tag: rank
        value: 17
    description: Clinical procedure performed on the subject in order to extract the
      biosample. ARGO mapping specimen::specimen_anatomic_location - Procedure::body_site
      ARGO mapping specimen::specimen_acquisition_interval - Procedure::time_performed
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.specimen_anatomic_location
    domain_of:
    - Measurement
    - Biosample
    - MedicalAction
    range: Procedure
  sampleProcessing:
    name: sampleProcessing
    annotations:
      rank:
        tag: rank
        value: 20
    description: Field to represent how the sample was processed. ARGO mapping specimen::specimen_processing
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.specimen_processing
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  sampleStorage:
    name: sampleStorage
    annotations:
      rank:
        tag: rank
        value: 21
    description: Field to represent how the sample was stored ARGO mapping specimen::specimen_storage
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.specimen_storage
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  sampleType:
    name: sampleType
    annotations:
      rank:
        tag: rank
        value: 6
    description: Recommended use of EFO term to describe the sample. e.g. Amplified
      DNA, ctDNA, Total RNA, Lung tissue, Cultured cells... ARGO mapping sample_registration::sample_type
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:sample_registration.sample_type
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  sampledTissue:
    name: sampledTissue
    annotations:
      rank:
        tag: rank
        value: 5
    description: 'UBERON class describing the tissue from which the specimen was collected.
      PDX-MI mapping: ''Specimen tumor tissue'' FHIR mapping: Specimen.type ARGO mapping
      sample_registration::specimen_tissue_source'
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:sample_registration.specimen_tissue_source
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  taxonomy:
    name: taxonomy
    annotations:
      rank:
        tag: rank
        value: 9
    description: NCBI taxonomic identifier (NCBITaxon) of the sample e.g. NCBITaxon:9606
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    domain_of:
    - Biosample
    - Individual
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  timeOfCollection:
    name: timeOfCollection
    annotations:
      rank:
        tag: rank
        value: 10
    description: An TimeElement describing either the age of the individual this biosample
      was derived from at the time of collection, or the time itself. See http://build.fhir.org/datatypes
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    domain_of:
    - Biosample
    range: TimeElement
    inlined: true
    inlined_as_list: true
  tumorGrade:
    name: tumorGrade
    annotations:
      rank:
        tag: rank
        value: 13
    description: Potentially a child term of NCIT:C28076 (Disease Grade Qualifier)
      or equivalent See https://www.cancer.gov/about-cancer/diagnosis-staging/prognosis/tumor-grade-fact-sheet
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  tumorProgression:
    name: tumorProgression
    annotations:
      rank:
        tag: rank
        value: 12
    description: Is the specimen tissue from the primary tumor, a metastasis or a
      recurrence? Most likely a child term of NCIT:C7062 (Neoplasm by Special Category)
      NCIT:C3677 (Benign Neoplasm) NCIT:C84509 (Primary Malignant Neoplasm) NCIT:C95606
      (Second Primary Malignant Neoplasm) NCIT:C3261 (Metastatic Neoplasm) NCIT:C4813
      (Recurrent Malignant Neoplasm)
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true

</code></pre>
</details>
<h3 id="induced">Induced</h3>
<details>
<pre><code class="language-yaml">name: Biosample
description: 'A Biosample refers to a unit of biological material from which the substrate
  molecules (e.g. genomic DNA, RNA, proteins) for molecular analyses (e.g. sequencing,
  array hybridisation, mass-spectrometry) are extracted. Examples would be a tissue
  biopsy, a single cell from a culture for single cell genome sequencing or a protein
  fraction from a gradient centrifugation. Several instances (e.g. technical replicates)
  or types of experiments (e.g. genomic array as well as RNA-seq experiments) may
  refer to the same Biosample. FHIR mapping: Specimen (http://www.hl7.org/fhir/specimen.html).'
from_schema: https://w3id.org/linkml/phenopackets/phenopackets
attributes:
  derivedFromId:
    name: derivedFromId
    annotations:
      rank:
        tag: rank
        value: 3
    description: The id of the parent biosample this biosample was derived from.
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    alias: derivedFromId
    owner: Biosample
    domain_of:
    - Biosample
    range: string
  description:
    name: description
    annotations:
      rank:
        tag: rank
        value: 4
    description: The biosample's description. This attribute contains human readable
      text. The "description" attributes should not contain any structured data.
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    alias: description
    owner: Biosample
    domain_of:
    - Cohort
    - ExternalReference
    - Measurement
    - PhenotypicFeature
    - Biosample
    - GeneDescriptor
    - VariationDescriptor
    range: string
  diagnosticMarkers:
    name: diagnosticMarkers
    annotations:
      rank:
        tag: rank
        value: 16
    description: Clinically relevant bio markers. Most of the assays such as IHC are
      covered by the NCIT under the sub-hierarchy NCIT:C25294 (Laboratory Procedure).
      e.g. NCIT:C68748 (HER2/Neu Positive), NCIT:C131711 (Human Papillomavirus-18
      Positive)
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    multivalued: true
    alias: diagnosticMarkers
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  files:
    name: files
    annotations:
      rank:
        tag: rank
        value: 18
    description: Pointer to the relevant file(s) for the biosample. Files relating
      to the entire individual e.g. a germline exome/genome should be associated with
      the Phenopacket rather than the Biosample it was derived from.
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    multivalued: true
    alias: files
    owner: Biosample
    domain_of:
    - Cohort
    - Family
    - Phenopacket
    - Biosample
    range: File
    inlined: true
    inlined_as_list: true
  histologicalDiagnosis:
    name: histologicalDiagnosis
    annotations:
      rank:
        tag: rank
        value: 11
    description: This is the pathologist’s diagnosis and may often represent a refinement
      of the clinical diagnosis given in the Patient/Clinical module. Should use the
      same terminology as diagnosis, but represent the pathologist’s findings. Normal
      samples would be tagged with the term "NCIT:C38757", "Negative Finding" ARGO
      mapping specimen::tumour_histological_type
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.tumour_histological_type
    rank: 1000
    alias: histologicalDiagnosis
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  id:
    name: id
    annotations:
      percent_encoded:
        tag: percent_encoded
        value: true
      rank:
        tag: rank
        value: 1
    description: biosamples SAMN08666232 Human Cell Atlas The Biosample id This is
      unique in the context of the server instance. ARGO mapping specimen::submitter_specimen_id
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.submitter_specimen_id
    alias: id
    owner: Biosample
    domain_of:
    - Cohort
    - Family
    - Phenopacket
    - ExternalReference
    - OntologyClass
    - Biosample
    - Allele
    - ChromosomeLocation
    - CopyNumber
    - Member
    - SequenceLocation
    - Text
    - VariationDescriptor
    - VcfRecord
    - Interpretation
    - Individual
    - Resource
    range: string
  individualId:
    name: individualId
    annotations:
      rank:
        tag: rank
        value: 2
    description: The id of the individual this biosample was derived from. ARGO mapping
      specimen::submitter_donor_id
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.submitter_donor_id
    rank: 1000
    alias: individualId
    owner: Biosample
    domain_of:
    - Biosample
    - Person
    range: string
  materialSample:
    name: materialSample
    annotations:
      rank:
        tag: rank
        value: 19
    description: 'This element can be used to specify the status of the sample. For
      instance, a status may be used as a normal control, often in combination with
      another sample that is thought to contain a pathological finding. We recommend
      use of ontology terms such as: EFO:0009654 (reference sample) or EFO:0009655
      (abnormal sample) ARGO mapping sample_registration::tumour_normal_designation'
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:sample_registration.tumour_normal_designation
    rank: 1000
    alias: materialSample
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  measurements:
    name: measurements
    annotations:
      rank:
        tag: rank
        value: 8
    description: ''
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    multivalued: true
    alias: measurements
    owner: Biosample
    domain_of:
    - Phenopacket
    - Biosample
    range: Measurement
    inlined: true
    inlined_as_list: true
  pathologicalStage:
    name: pathologicalStage
    annotations:
      rank:
        tag: rank
        value: 14
    description: ARGO mapping specimen::pathological_tumour_staging_system ARGO mapping
      specimen::pathological_stage_group
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.pathological_tumour_staging_system
    rank: 1000
    alias: pathologicalStage
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  pathologicalTnmFinding:
    name: pathologicalTnmFinding
    annotations:
      rank:
        tag: rank
        value: 15
    description: ARGO mapping specimen::pathological_t_category ARGO mapping specimen::pathological_n_category
      ARGO mapping specimen::pathological_m_category
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.pathological_t_category
    rank: 1000
    multivalued: true
    alias: pathologicalTnmFinding
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  phenotypicFeatures:
    name: phenotypicFeatures
    annotations:
      rank:
        tag: rank
        value: 7
    description: Phenotypic characteristics of the BioSample, for example histological
      findings of a biopsy.
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    multivalued: true
    alias: phenotypicFeatures
    owner: Biosample
    domain_of:
    - Phenopacket
    - Biosample
    range: PhenotypicFeature
    inlined: true
    inlined_as_list: true
  procedure:
    name: procedure
    annotations:
      rank:
        tag: rank
        value: 17
    description: Clinical procedure performed on the subject in order to extract the
      biosample. ARGO mapping specimen::specimen_anatomic_location - Procedure::body_site
      ARGO mapping specimen::specimen_acquisition_interval - Procedure::time_performed
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.specimen_anatomic_location
    alias: procedure
    owner: Biosample
    domain_of:
    - Measurement
    - Biosample
    - MedicalAction
    range: Procedure
  sampleProcessing:
    name: sampleProcessing
    annotations:
      rank:
        tag: rank
        value: 20
    description: Field to represent how the sample was processed. ARGO mapping specimen::specimen_processing
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.specimen_processing
    rank: 1000
    alias: sampleProcessing
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  sampleStorage:
    name: sampleStorage
    annotations:
      rank:
        tag: rank
        value: 21
    description: Field to represent how the sample was stored ARGO mapping specimen::specimen_storage
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:specimen.specimen_storage
    rank: 1000
    alias: sampleStorage
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  sampleType:
    name: sampleType
    annotations:
      rank:
        tag: rank
        value: 6
    description: Recommended use of EFO term to describe the sample. e.g. Amplified
      DNA, ctDNA, Total RNA, Lung tissue, Cultured cells... ARGO mapping sample_registration::sample_type
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:sample_registration.sample_type
    rank: 1000
    alias: sampleType
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  sampledTissue:
    name: sampledTissue
    annotations:
      rank:
        tag: rank
        value: 5
    description: 'UBERON class describing the tissue from which the specimen was collected.
      PDX-MI mapping: ''Specimen tumor tissue'' FHIR mapping: Specimen.type ARGO mapping
      sample_registration::specimen_tissue_source'
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    exact_mappings:
    - ARGO:sample_registration.specimen_tissue_source
    rank: 1000
    alias: sampledTissue
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  taxonomy:
    name: taxonomy
    annotations:
      rank:
        tag: rank
        value: 9
    description: NCBI taxonomic identifier (NCBITaxon) of the sample e.g. NCBITaxon:9606
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    alias: taxonomy
    owner: Biosample
    domain_of:
    - Biosample
    - Individual
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  timeOfCollection:
    name: timeOfCollection
    annotations:
      rank:
        tag: rank
        value: 10
    description: An TimeElement describing either the age of the individual this biosample
      was derived from at the time of collection, or the time itself. See http://build.fhir.org/datatypes
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    alias: timeOfCollection
    owner: Biosample
    domain_of:
    - Biosample
    range: TimeElement
    inlined: true
    inlined_as_list: true
  tumorGrade:
    name: tumorGrade
    annotations:
      rank:
        tag: rank
        value: 13
    description: Potentially a child term of NCIT:C28076 (Disease Grade Qualifier)
      or equivalent See https://www.cancer.gov/about-cancer/diagnosis-staging/prognosis/tumor-grade-fact-sheet
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    alias: tumorGrade
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true
  tumorProgression:
    name: tumorProgression
    annotations:
      rank:
        tag: rank
        value: 12
    description: Is the specimen tissue from the primary tumor, a metastasis or a
      recurrence? Most likely a child term of NCIT:C7062 (Neoplasm by Special Category)
      NCIT:C3677 (Benign Neoplasm) NCIT:C84509 (Primary Malignant Neoplasm) NCIT:C95606
      (Second Primary Malignant Neoplasm) NCIT:C3261 (Metastatic Neoplasm) NCIT:C4813
      (Recurrent Malignant Neoplasm)
    from_schema: https://w3id.org/linkml/phenopackets/phenopackets
    rank: 1000
    alias: tumorProgression
    owner: Biosample
    domain_of:
    - Biosample
    range: OntologyClass
    inlined: true
    inlined_as_list: true

</code></pre>
</details>
</article>
</div>
<script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>
</div>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
<script src="../assets/javascripts/bundle.5a2dcb6a.min.js"></script>
<script src="https://unpkg.com/mermaid@9.3.0/dist/mermaid.min.js"></script><script>mermaid.initialize({});</script></body>
</html>